modules/network-observability-tcp-flag-syn-flood.adoc

```diff
@@ -0,0 +1,91 @@
+kind: FlowMetric
+metadata:
+  name: flows-with-flags-per-destination
+spec:
+  metricName: flows_with_flags_per_destination_total
+  type: Counter
+  labels: [SrcSubnetLabel,DstSubnetLabel,DstK8S_Name,DstK8S_Type,DstK8S_HostName,DstK8S_Namespace,Flags]
+----
+====
++
+.FlowMetric configuration to count flows per source host and resource, with TCP flags
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: flows.netobserv.io/v1alpha1
+kind: FlowMetric
+metadata:
+  name: flows-with-flags-per-source
+spec:
+  metricName: flows_with_flags_per_source_total
+  type: Counter
+  labels: [DstSubnetLabel,SrcSubnetLabel,SrcK8S_Name,SrcK8S_Type,SrcK8S_HostName,SrcK8S_Namespace,Flags]
+----
+====
+
+. Deploy the following `AlertingRule` to alert for SYN flooding:
++
+.Alert for SYN flooding
+[source,yaml]
+----
+apiVersion: monitoring.openshift.io/v1
+kind: AlertingRule
+metadata:
+  name: netobserv-syn-alerts
+  namespace: openshift-monitoring
+# ...
+  spec:
+  groups:
+  - name: NetObservSYNAlerts
+    rules:
+    - alert: NetObserv-SYNFlood-in
+      annotations:
+        message: |-
+          {{ $labels.job }}: incoming SYN-flood attack suspected to Host={{ $labels.DstK8S_HostName}}, Namespace={{ $labels.DstK8S_Namespace }}, Resource={{ $labels.DstK8S_Name }}. This is characterized by a high volume of SYN-only flows with different source IPs and/or ports.
+        summary: "Incoming SYN-flood"
+      expr: sum(rate(netobserv_flows_with_flags_per_destination_total{Flags="2"}[1m])) by (job, DstK8S_HostName, DstK8S_Namespace, DstK8S_Name) > 300      <1>
+      for: 15s
+      labels:
+        severity: warning
+        app: netobserv
+    - alert: NetObserv-SYNFlood-out
+      annotations:
+        message: |-
+          {{ $labels.job }}: outgoing SYN-flood attack suspected from Host={{ $labels.SrcK8S_HostName}}, Namespace={{ $labels.SrcK8S_Namespace }}, Resource={{ $labels.SrcK8S_Name }}. This is characterized by a high volume of SYN-only flows with different source IPs and/or ports.
+        summary: "Outgoing SYN-flood"
+      expr: sum(rate(netobserv_flows_with_flags_per_source_total{Flags="2"}[1m])) by (job, SrcK8S_HostName, SrcK8S_Namespace, SrcK8S_Name) > 300       <1>
+      for: 15s
+      labels:
+        severity: warning
+        app: netobserv
+# ...
+----
+<1> In this example, the threshold for the alert is `300`; however, you can adapt this value empirically. A threshold that is too low might produce false-positives, and if it's too high it might miss actual attacks. It is to be determined empirically.
+
+.Verification
+. In the web console, click *Manage Columns* in the *Network Traffic* table view and click *TCP flags*.
+. In the *Network Traffic* table view, filter on *TCP protocol SYN TCPFlag*. Large number of flows with same *byteSize* indicate a SYN flood.
+. Go to *Observe* -> *Alerting* -> *Alerting Rules* tab and filter on *netobserv-synflood-in alert*. The alert should fire when SYN flooding occurs. 
```

Comment:
I feel like the GUI direction here is a bit off, and I can't tell what the right answer would be without the GUI in front of me. Is it that you go to the Observe page, then the Alerting page, and then select the Alerting Rules tab? Or are all three of those different tabs? 

I'm not sure what the best answer would be here, but either removing tab or breaking up that set of three GUI stops might be the accurate description:
```suggestion
. Go to *Observe* -> *Alerting* -> *Alerting Rules* and filter on *netobserv-synflood-in alert*. The alert should fire when SYN flooding occurs. 
```

```suggestion
. Go to *Observe* -> *Alerting* and select the *Alerting Rules* tab, then filter on *netobserv-synflood-in alert*. The alert should fire when SYN flooding occurs. 
```

Reply:
I think the second suggestion makes sense. "Observe > Alerting" is just a sub-menu item to navigate to. "Alerting Rules" is a tab to click on.